src/XIVLauncher.Core/WineManager.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Serilog;
+using XIVLauncher.Common;
+using XIVLauncher.Common.Unix.Compatibility;
+
+namespace XIVLauncher.Core;
+
+public enum WineType
+{
+    [SettingsDescription("Managed by XIVLauncher", "Choose a patched version of wine made specifically for XIVLauncher")]
+    Managed,
+
+    [SettingsDescription("Custom", "Point XIVLauncher to a custom location containing wine binaries to run the game with.")]
+    Custom,
+}
+
+public enum WineVersion
+{
+    [SettingsDescription("Wine-xiv 8.5", "A patched version of Wine-staging 8.5. The current default.")]
+    Wine8_5,
+
+    [SettingsDescription("Wine-xiv 7.10", "A legacy patched version of Wine, based on 7.10. A previous default")]
+    Wine7_10,
+}
+
+public static class WineManager
+{
+#if WINE_XIV_ARCH_LINUX
+    private const string DISTRO = "arch";
+#elif WINE_XIV_FEDORA_LINUX
+    private const string DISTRO = "fedora";
+#else
+    private const string DISTRO = "ubuntu";
+#endif
```

Comment:
Would it be possible for us to determine this without setting flags at build time so we only need to produce a single artifact?

Reply:
Actually, yes. On just about any system, we can read the /etc/os-release file, and the first line will be NAME="some distro". So we could search for Arch and Fedora/SUSE, and then give ubuntu if it doesn't match those.

Reply:
Perfect, would be nice if we could get that change in with this if possible

Reply:
Alright, I've added a function for this. Basically, fedora has an /etc/fedora-release file, arch derivatives have an /etc/arch-release file, and everyone else can get the ubuntu release.